daemon/cluster/listen_addr.go

```diff
@@ -123,6 +123,22 @@ func validateDefaultAddrPool(defaultAddrPool []string, size uint32) error {
 	return nil
 }
 
+// validateDataPathPort validates vxlan udp port (data path port) number.
+func validateDataPathPort(portNum uint32) error {
+	// if the value comes as 0 by any reason we set it to default value 4789
+	if portNum == 0 {
+		portNum = 4789
+	}
+	// IANA procedures for each range in detail
+	// The Well Known Ports, aka the System Ports, from 0-1023
+	// The Registered Ports, aka the User Ports, from 1024-49151
+	// The Dynamic Ports, aka the Private Ports, from 49152-65535
+	// So we can allow range between 1024 to 49151
+	if portNum < 1024 || portNum > 49151 {
+		return fmt.Errorf("validateDataPathPort : Data Path Port number is not in valid range %d", portNum)
```

Comment:
Can you;

- remove the `validateDataPathPort:` here? The error message should only describe the error condition (not what function did the validation; it's not a debug-message).
- `Path` and `Port` should be lowercase (and errors should not start with an uppercase)
- Add the _valid_ range in the error-message (this will help the user resolve  the problem if they specify an invalid number).

Perhaps;

```suggestion
		return fmt.Errorf("data path port number is not in valid range (1024-49151): %d", portNum)
```